src/services/BidService.ts

```diff
@@ -0,0 +1,121 @@
+import JobContext from 'decentraland-gatsby/dist/entities/Job/context'
+
+import BidModel from '../entities/Bid/model'
+import { BidStatus, UnpublishedBid } from '../entities/Bid/types'
+import { ProposalType } from '../entities/Proposal/types'
+import { DEFAULT_CHOICES } from '../entities/Proposal/utils'
+import Time from '../utils/date/Time'
+
+import { ErrorService } from './ErrorService'
+import { ProposalService } from './ProposalService'
+
+const MINIMUM_BIDS_TO_PUBLISH = Number(process.env.MINIMUM_BIDS_TO_PUBLISH)
+export default class BidService {
+  static async createBid(
+    linked_proposal_id: string,
+    author_address: string,
+    bid_proposal_data: UnpublishedBid['bid_proposal_data']
+  ) {
+    const bidsSubmissionWindow = process.env.SUBMISSION_WINDOW_DURATION_BID
+      ? Number(process.env.SUBMISSION_WINDOW_DURATION_BID)
+      : undefined
+
+    if (!bidsSubmissionWindow) {
+      throw new Error('Bids submission window is not defined')
+    }
+    if (await this.isSubmissionWindowFinished(linked_proposal_id)) {
+      throw new Error('Bids submission window is finished')
+    }
+    const tenderBids = await BidModel.getBidsInfoByTender(linked_proposal_id)
+
+    if (tenderBids.find((bid) => bid.author_address === author_address)) {
+      throw new Error('Bid already exists')
+    }
+
+    const publish_at =
+      tenderBids.length > 0 ? tenderBids[0].publish_at : Time().add(bidsSubmissionWindow, 'seconds').toISOString()
+    await BidModel.createBid({
+      linked_proposal_id,
+      bid_proposal_data,
+      author_address,
+      publish_at,
+      status: BidStatus.Pending,
+    })
+  }
+
+  static async publishBids(context: JobContext) {
+    const pendingBids = await BidModel.getBidsReadyToPublish()
+    if (pendingBids.length === 0) {
+      return
+    }
+
+    if (isNaN(MINIMUM_BIDS_TO_PUBLISH) || MINIMUM_BIDS_TO_PUBLISH === 0) {
```

Comment:
Consider having a default for MINIMUM_BIDS_TO_PUBLISH initialization instead of this NaN check 

`const MINIMUM_BIDS_TO_PUBLISH = Number(process.env.MINIMUM_BIDS_TO_PUBLISH || 0)`